core/templates/domain/objects/NumberWithUnitsObjectFactory.ts

```diff
@@ -150,7 +150,8 @@ export class NumberWithUnitsObjectFactory {
           units = rawInput.substr(ind).trim();
         }
 
-        var keys = Object.keys(ObjectsDomainConstants.CURRENCY_UNITS);
+        var keys = <(keyof typeof ObjectsDomainConstants.CURRENCY_UNITS)[]>
+          Object.keys(ObjectsDomainConstants.CURRENCY_UNITS);
```

Comment:
Would this work?
```suggestion
        var keys: (keyof typeof ObjectsDomainConstants.CURRENCY_UNITS)[] = (
          Object.keys(ObjectsDomainConstants.CURRENCY_UNITS));
```

Reply:
I just tried inserting this code, and it gave the following linting error:

error TS2322: Type 'string[]' is not assignable to type '("dollar" | "rupee" | "cent" | "paise")[]'.

 Type 'string' is not assignable to type '"dollar" | "rupee" | "cent" | "paise"'.